nipype/algorithms/confounds.py

```diff
@@ -613,13 +663,19 @@ def _list_outputs(self):
                 save_pre_filter = os.path.abspath('pre_filter.tsv')
             outputs['pre_filter_file'] = save_pre_filter
 
+        save_metadata = self.inputs.save_metadata
+        if save_metadata:
+            if isinstance(save_metadata, bool):
```

Comment:
remove this line

Reply:
Hmm, looking at it now, I'm not sure that change would work because the conditional branch should be entered if `save_metadata` is either `True` or a string (the metadata file path) so as to set the output path.
```python
        if save_metadata:
            if isinstance(save_metadata, bool):
                save_metadata = os.path.abspath('component_metadata.tsv')
            outputs['metadata_file'] = save_metadata
```
Of course, we could circumvent this entirely with a `SimpleInterface` as per your earlier suggestion. I'm not sure if there's a good reason for not using `SimpleInterface` or `fname_presuffix` here (perhaps this function predates their integration into `nipype`?), but I mirrored this block structure from the `save_pre_filter` block above it. I could definitely try and see if I can refactor into a `SimpleInterface` with `fname_presuffix` if you think it would be worthwhile.